BackofficeBundle/Manager/RouteDocumentManager.php

```diff
@@ -159,9 +170,30 @@ protected function completeRoutePattern($parentId = null, $suffix = null, $langu
      */
     public function clearForNode(NodeInterface $node)
     {
-        return $this->routeDocumentRepository->findByNodeIdSiteIdAndLanguage($node->getNodeId(), $node->getSiteId(), $node->getLanguage());
+        $nodes = $this->getTreeNode($node);
+        $routes = array();
+
+        foreach ($nodes as $node) {
+            $routes = array_merge($this->routeDocumentRepository->findByNodeIdSiteIdAndLanguage($node->getNodeId(), $node->getSiteId(), $node->getLanguage()), $routes);
+        }
+
+        return $routes;
+    }
+
+    protected function getTreeNode(NodeInterface $node, $site = null)
+    {
+        $nodes = array($node);
+        $site = (null === $site) ? $this->siteRepository->findOneBySiteId($node->getSiteId()) : $site;
```

Comment:
as you wish. I prefer this notation.
you want :  
if (null === $site) {
$site = $this->siteRepository->findOneBySiteId($node->getSiteId())
}
3 lines!
@itkg-ngilain